storage/ActiongraphImpl.cc

```diff
@@ -130,6 +130,23 @@ namespace storage
     }
 
 
+    void
+    Actiongraph::Impl::add_chain(const vector<vector<vertex_descriptor>>& vert_vectors)
+    {
+	vector<vector<Actiongraph::Impl::vertex_descriptor>> non_empty_vectors;
+	for (const vector<vertex_descriptor>& vector : vert_vectors)
+	    if (!vector.empty()) non_empty_vectors.push_back(vector);
+
+	if (non_empty_vectors.size() > 1)
+	    for (size_t i = 1; i < non_empty_vectors.size(); ++i)
```

Comment:
Right now it's used for dependencies of operations within a LVM VG. The cost should be kind of obvious from the description, but I will add a note.